src/main/kotlin/org/rust/ide/annotator/RsErrorAnnotator.kt

```diff
@@ -884,6 +885,29 @@ class RsErrorAnnotator : AnnotatorBase(), HighlightRangeExtension {
         RsDiagnostic.LoopOnlyKeywordUsedOutsideOfLoopError(expr.operator).addToHolder(holder)
     }
 
+    private fun checkArrayType(holder: RsAnnotationHolder, o: RsArrayType) {
+        collectDiagnostics(holder, o)
+        val sizeExpr = o.expr
+        if (sizeExpr != null && o.arraySize == null) {
+            checkArraySizeExpr(holder, sizeExpr)
+        }
+    }
+    private fun checkArrayExpr(holder: RsAnnotationHolder, o: RsArrayExpr) {
+        val sizeExpr = o.sizeExpr
+        if (sizeExpr != null) {
+            checkArraySizeExpr(holder, sizeExpr)
+        }
+    }
+
+    private fun checkArraySizeExpr(holder: RsAnnotationHolder, sizeExpr: RsExpr) {
+        sizeExpr.descendantsOfTypeOrSelf<RsPathExpr>().forEach {
+            val ref = it.path.reference?.resolve()
+            if (ref != null && ref !is RsConstant && (ref as? RsFunction)?.isConst != true) {
+                RsDiagnostic.NonConstantValueInConstantError(it).addToHolder(holder)
```

Comment:
BTW compiler produces [E0015](https://doc.rust-lang.org/error-index.html#E0015) when you try to use non-constant value and it's not a local variable.
Looks like it's quite easy to check it here so let's produce the same errors like compiler